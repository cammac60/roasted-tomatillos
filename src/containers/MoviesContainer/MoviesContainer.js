import React, { Component } from 'react';
import { connect } from 'react-redux';
import './MoviesContainer.scss';
import SmallMovieCard from '../../components/SmallMovieCard/SmallMovieCard';
import LoadingImage from '../../components/LoadingImage/LoadingImage';

export class MoviesContainer extends Component {


  createCards = dataset => {
    return dataset.map(movie => (
      <SmallMovieCard
        key={movie.id}
        id={movie.id}
        img={movie.poster_path}
        rate={movie.average_rating}
        title={movie.title} />
    ));
  }

  render() {
    const { movies, isLoaded, error } = this.props;
    const notification = (error !== '')
      ? <h2>{error}</h2>
      : <LoadingImage />

    return (
      <main className="movies-container">
        { (isLoaded && error === '')
            ? this.createCards(movies)
            : notification }
      </main>
    )
  }
}

export const mapStateToProps = ({movies, isLoaded, error}) => ({
  movies,
  isLoaded,
  error
})

export default connect(mapStateToProps)(MoviesContainer);
