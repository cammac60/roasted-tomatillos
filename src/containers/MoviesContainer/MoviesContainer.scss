@import "../../variables.scss";

.movies-container {
  display: flex;
  flex-wrap: wrap;
  height: 80vh;
  overflow: scroll;
  margin: 20px;
  padding: 20px;
  h2 {
    color: #fff;
    margin: auto;
    background-color: $main-background-color-transparent;
    backdrop-filter: blur(4px);
    border: 1px solid $main-accent-color;
    padding: 40px;
  }
}
