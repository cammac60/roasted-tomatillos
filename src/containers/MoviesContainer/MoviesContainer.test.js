import React from 'react';
import { MoviesContainer, mapStateToProps } from './MoviesContainer';
import { shallow } from 'enzyme';

describe("MoviesContainer", () => {
  const mockMoviesData =
    [{
      id: 1,
      title: "Jumanji: The Next Level",
      poster_path: 'https://some.url/poster-image',
      backdrop_path: 'https://some.url/bg-image',
      release_date: '2019-12-04',
      overview: 'In Jumanji: The Next Level, the gang is back but the game has changed.',
      average_rating: 4
    }];

  describe("MoviesContainer component", () => {
    let container, instance;

    const store = {
      movies: mockMoviesData,
      isLoaded: true,
      error: '',
    }

    beforeEach(() => {
      container = shallow(
        <MoviesContainer {...store} />
      );

      instance = container.instance();
    });

    it("should match snapshot if data was loaded and no error", () => {
      expect(container).toMatchSnapshot();
    });

    it("should match snapshot if data is loading", () => {
      const containerLoading = shallow(
        <MoviesContainer {...store} isLoaded={false} />
      );
      expect(containerLoading).toMatchSnapshot();
    });

    it("should match snapshot if there is error after fetch", () => {
      const containerLoading = shallow(
        <MoviesContainer {...store} error="Failed to fetch" />
      );
      expect(containerLoading).toMatchSnapshot();
    });

    it("should call createCards method after rendering", () => {
      const spy = jest.spyOn(instance, 'createCards')
        .mockImplementation(() => {});
      instance.forceUpdate()

      expect(spy).toHaveBeenCalled();
    });

  });

  describe('mapStateToProps', () => {
    it('should return an object with the movies array, isLoaded and error', () => {
      // Setup
      const mockState = {
        movies: mockMoviesData,
        isLoaded: true,
        error: '',
        user: {name: 'Ray'}
      };

      const expected = {
        movies: mockMoviesData,
        isLoaded: true,
        error: ''
      };

      // Execution
      const mappedProps = mapStateToProps(mockState);

      // Expectation
      expect(mappedProps).toEqual(expected);
    });
  });
});
