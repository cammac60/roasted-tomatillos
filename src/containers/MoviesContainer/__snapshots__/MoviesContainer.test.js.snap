// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer MoviesContainer component should match snapshot if data is loading 1`] = `
<main
  className="movies-container"
>
  <LoadingImage />
</main>
`;

exports[`MoviesContainer MoviesContainer component should match snapshot if data was loaded and no error 1`] = `
<main
  className="movies-container"
>
  <SmallMovieCard
    id={1}
    img="https://some.url/poster-image"
    key="1"
    rate={4}
    title="Jumanji: The Next Level"
  />
</main>
`;

exports[`MoviesContainer MoviesContainer component should match snapshot if there is error after fetch 1`] = `
<main
  className="movies-container"
>
  <h2>
    Failed to fetch
  </h2>
</main>
`;
